pkg/sensors/tracing/generickprobe.go

```diff
@@ -253,6 +253,53 @@ func createMultiKprobeSensor(sensorPath string, multiIDs, multiRetIDs []idtable.
 	return progs, maps
 }
 
+// preValidateKprobes pre-validates the semantics and BTF information of a Kprobe spec
+//
+// Pre validate the kprobe semantics and BTF information in order to separate
+// the kprobe errors from the BPF related ones.
+func preValidateKprobes(name string, kprobes []v1alpha1.KProbeSpec) error {
+	for i := range kprobes {
+		f := &kprobes[i]
+
+		// modifying f.Call directly since BTF validation
+		// later will use v1alpha1.KProbeSpec object
+		if f.Syscall {
+			prefixedName, err := arch.AddSyscallPrefix(f.Call)
+			if err != nil {
+				logger.GetLogger().WithFields(logrus.Fields{
+					"sensor": name,
+				}).WithError(err).Warn("Kprobe spec pre-validation of syscall prefix failed")
+			} else {
+				f.Call = prefixedName
+			}
+		}
+
+		// Now go over BTF validation
+		btfobj, err := btf.NewBTF()
+		if err != nil {
+			return err
+		}
+
+		if err := btf.ValidateKprobeSpec(btfobj, f); err != nil {
+			if warn, ok := err.(*btf.ValidationWarn); ok {
+				logger.GetLogger().WithFields(logrus.Fields{
+					"sensor": name,
+				}).WithError(warn).Warn("Kprobe spec pre-validation failed")
+			} else if e, ok := err.(*btf.ValidationFailed); ok {
+				return fmt.Errorf("kprobe spec pre-validation failed: %w", e)
+			} else {
+				logger.GetLogger().Warnf("invalid or old kprobe spec: %s", err)
+			}
+		}
+
+		logger.GetLogger().WithFields(logrus.Fields{
+			"sensor": name,
+		}).Debug("Kprobe spec pre-validation succeeded")
```

Comment:
Let's remove or add an `else` because as it is now we can see both messages "kprobe spec pre-valudation failed" and "Kprobe spec pre-validatoin succeeded" for the same sensor.